04 auth

Alex-T-T opened the pull request.

Alex-T-T committed:
auth first steps

1) add authrouter
2) create userSchema
3) create authController
4) add userRegisterValidation
5) install bcryptjs

Alex-T-T committed:
login

1) update registerController - add hash to password
2) add loginController

Alex-T-T committed:
add getCurrentUser

1) add currentUserController
2) add auth.js in folder "middlewares"

Alex-T-T committed:
logout

1) add logout
2) add pagination

Alex-T-T committed:
filtration & update user.subscription

1) add filtration for favourite contacts
2) add update for User "subscription" field

Alex-T-T committed:
update favourite

Alex-T-T committed:
subscription validation

add updateContactFavouriteValidation

OnyshchenkoO commented:
Поправ усюди табуляцію і виправ помилки еслінту, бо код дуже важко читати

OnyshchenkoO commented on src/controllers/authController.js:
```diff
@@ -0,0 +1,74 @@
+}
+
+const updateUserSubscriptionController = async (req, res, next) => {
+    console.log(req.body);
```
Прибери це

Alex-T-T replied:
прибрав

OnyshchenkoO commented on src/middlewares/auth.js:
```diff
@@ -0,0 +1,33 @@
+const jwt = require('jsonwebtoken');
+const { SECRET_KEY } = process.env;
+
+
+
+
```
Вони тут не потрібні

Alex-T-T replied:
видалив

OnyshchenkoO commented on src/middlewares/validationMiddlware.js:
```diff
@@ -56,10 +56,73 @@ module.exports = {
+    },
+
+    userRegisterValidation: (req, res, next) => {
+        const schema = Joi.object({
+            password: Joi.string()
+            .required(),
+
+        email: Joi.string()
+            .email({minDomainSegments: 2, tlds: {allow: ['com', 'net']}})
+            .required(),
+
+        subscription: Joi.string()
+            .optional(),
```
Винеси всі схеми в окремі файл

Alex-T-T replied:
виніс кожну окремо в файл та згрупував в окремій папці validationMiddlewares

OnyshchenkoO commented on src/routes/api/contacts.js:
```diff
@@ -1,31 +1,33 @@
+router.get('/', auth, controllerCheck(getAllContactsController));
 
 // GET contact by ID
 router.get('/:contactId', isValidId, controllerCheck(getContactController));
```
Тут теж має бути авторизація. Хоч цього в ДЗ і не сказано, але в нас контакти тепер мають owner'a і лише він може їх редагувати\видаляти і т.д

Alex-T-T replied:
додав

OnyshchenkoO commented on src/routes/api/contacts.js:
```diff
@@ -1,31 +1,33 @@
+router.post('/', auth, addContactValidation, controllerCheck(postContactController));
 
 // DELETE - remove contact by ID
 router.delete('/:contactId', isValidId, controllerCheck(deleteContactController));
```
Тут теж має бути авторизація

Alex-T-T replied:
додав

OnyshchenkoO commented on src/routes/api/contacts.js:
```diff
@@ -1,31 +1,33 @@
 router.delete('/:contactId', isValidId, controllerCheck(deleteContactController));
 
 // PUT - update contact by ID
 router.put('/:contactId', isValidId, putContactValidation, controllerCheck(putContactController));
```
Тут теж має бути авторизація

Alex-T-T replied:
додав

OnyshchenkoO commented on src/routes/api/contacts.js:
```diff
@@ -1,31 +1,33 @@
-// PATCH - update contact field 'favorite' by contact ID
-router.patch('/:contactId/favorite', isValidId, updateContactFavoriteValidation, controllerCheck(updateContactFavoriteController) ) 
+// PATCH - update contact field 'favourite' by contact ID
+router.patch('/:contactId/favourite', isValidId, updateContactFavouriteValidation, controllerCheck(updateContactFavouriteController) ) 
```
Тут теж має бути авторизація

Alex-T-T replied:
додав

Alex-T-T committed:
add auth to some routers

1) add auth to  routers:

Alex-T-T committed:
replace validation middlewares

create new own files for everyone validation Joi schema at folder "validationMiddlewares"
